apps/testingaccessibility/src/components/app/navigation.tsx

```diff
@@ -10,33 +10,21 @@ import toast from 'react-hot-toast'
 import Link from 'next/link'
 import cx from 'classnames'
 import Image from 'next/image'
-import {useUser} from '../../context/user-context'
-import {getAbilityFromToken} from '../../server/ability'
-import {useQuery} from 'react-query'
+import {useCurrentAbility} from '../../hooks/use-current-ability'
 
-// TODO: This is a temporary solution to get the ability from the token
-//       It's named poorly and is clumping stuff together, but it works!
-function useRefinedSession() {
-  const {data, status} = useSession()
-  const {user} = useUser()
+function useNavState() {
+  const {ability, status} = useCurrentAbility()
 
-  const isSignedIn = Boolean(data?.user)
-  const isLoadingUser = status === 'loading'
-
-  // this is all client side so it isn't secure secure but for display
-  // purposes it's fine
-  const {data: ability} = useQuery(
-    ['ability', user],
-    async () => await getAbilityFromToken(user),
+  console.log(
+    `DesktopNav: can invite team members: ${ability?.can('invite', 'Team')}`,
+  )
+  console.log(
+    `DesktopNav: can view content: ${ability?.can('view', 'Content')}`,
   )
 
   return {
-    isSignedIn,
-    isLoadingUser,
-    data,
-    status,
-    user,
-    ability,
+    isSignedIn: status === 'authenticated',
+    isLoadingUser: status === 'loading',
   }
 }
```

Comment:
`useNavState` is *specific* to this component with some handy booleans for layout purposes below. Currently they are only based on status, but could also utilize abilities